### DISCUSSION TITLE
New fields for FAP Review of Proposals

### DISCUSSION THREAD

**Discussion by gnyiri** [UserOfficeProject/user-office-core#651] [Ideas]
Hi there,

at ELI-ERIC we would need some new attributes for FAP Reviews. Just like for discussion #645, we would prefer configurable FAP review fields (through templates). AFAIK, STFC would also be interesed in this feature. 

Gergely

**Comment by gnyiri** [comment]
Hi,
I started the implementation by adding new template_category, template_group, question_datatype (FAP Review basis), ..

What do you think, shall I move existing fields 'comment' and 'grade' to the new, 'FAP review basis' question or shall I keep them in the FAP review table?
Thanks,
Gergely

**Reply by martin-trajanovski** [reply]
In my opinion they should be moved in a 'FAP review basis' question and data migration should be done to move the existing data later.

**Reply by gnyiri** [reply]
thanks. At least 'grade' seems to be quite essential part of the review: some frontend components (e.g. table on ReviewPage) has 'Grade' or 'Average Grade' column. Thus, moving grade to a template would require more changes than I originally thought (e.g. getReviews endpoint and/or the frontend code where grade is queried/updated). 

What do you think about keeping at least grade in the review? I guess that this case is similar to those with the abstract field of the proposal (which is templated already, but abstract was kept in proposal table).

**Reply by martin-trajanovski** [reply]
Yes I agree. If this reduces some changes let's keep it like that. It can be part of the template as fixed question but we can keep it in the same database table and avoid plenty of refactor.
